### PR TITLE
doc: Add references for the generator/constant used in Bech32(m)

### PR DESCRIPTION
**Pull request by sipa**
I often find myself recreating this, or looking up references for this construction. So instead, this seems like as good a place as any to place a summary.

### CODE REVIEW

**Comment by tryphe** [comment]
ACK 260ae257f0edb708b30d742db29d1caf37a3b2e7

**Review by arik-so — APPROVED** [review]

**Comment by Zero-1729** [comment]
Concept ACK

File: src/bech32.cpp [hunk 1]
```diff
@@ -66,6 +66,26 @@ uint32_t PolyMod(const data& v)
...
+    // E_mod = x**2 + F.fetch_int(9)*x + F.fetch_int(23)
+    // E.<e> = F.extension(E_mod) # GF(1024) extension field definition
+    // for p in divisors(E.order() - 1): # Verify e has order 1023.
+    //    assert((e**p == 1) == (p % 1023 == 0))
+    // G = lcm([(e**i).minpoly() for i in range(997,1000)])
+    // print(G) # Print out the generator
+    //
+    // It demonstrates that g(x) is the least common multiple of the minimal polynomials
+    // of 3 consecutive powers (997,998,999) of a primitive element (e) of GF(1024).
+    // That guarantees it is in fact the generator of a primitive BCH code with cycle
```
Comment (Zero-1729) [inline src/bech32.cpp:86]: Minor grammar nit, feel free to ignore.

```suggestion
    // That guarantees it is, in fact, the generator of a primitive BCH code with cycle
```

Comment (sipa) [reply src/bech32.cpp]: Done.

**Review by Zero-1729 — COMMENTED** [review]
crACK b8cd2a429271b7af12e98186449ddc00fc7580e6

**Pull request merged**